### PR TITLE
SALTO-2385: Extract references from workflow status properties

### PR DESCRIPTION
**Pull request by alonstern**
Parsed references from workflow status properties

---
_Release Notes_: 
_Jira Adapter_:
- Parsed references from properties values inside workflow statuses

---
_User Notifications_: 
_Jira Adapter_:
- Parsed references from properties values inside workflow statuses

### CODE REVIEW

File: packages/jira-adapter/src/filters/workflow/workflow_properties_filter.ts [hunk 1]
```diff
@@ -46,68 +48,90 @@ const convertStatusesPropertiesToMap = (instance: WorkflowInstance): void => {
...
-    elements.push(propertyType)
+      elements.push(propertyType)
 
-    const workflowStatusType = findObject(elements, 'WorkflowStatus')
-    if (workflowStatusType !== undefined) {
-      workflowStatusType.fields.properties = new Field(workflowStatusType, 'properties', new ListType(propertyType), { [CORE_ANNOTATIONS.REQUIRED]: true })
-    }
+      const workflowStatusType = findObject(elements, 'WorkflowStatus')
+      if (workflowStatusType !== undefined) {
+        workflowStatusType.fields.properties = new Field(workflowStatusType, 'properties', new ListType(propertyType), { [CORE_ANNOTATIONS.REQUIRED]: true })
```
Comment (netama) [inline packages/jira-adapter/src/filters/workflow/workflow_properties_filter.ts:75]: nit: what's the current type? maybe we can override this in the infra?

File: packages/jira-adapter/src/filters/workflow/workflow_properties_filter.ts [hunk 2]
```diff
@@ -46,68 +48,90 @@ const convertStatusesPropertiesToMap = (instance: WorkflowInstance): void => {
...
+
+      _.pullAll(changes, relevantChanges)
+      changesToReturn.forEach(change => changes.push(change))
+    },
+
+    onDeploy: async changes => {
+      const relevantChanges = changes
+        .filter(isInstanceChange)
+        .filter(change => getChangeData(change).elemID.typeName === WORKFLOW_TYPE_NAME)
+        .filter(change => isWorkflowInstance(getChangeData(change)))
```
Comment (netama) [inline packages/jira-adapter/src/filters/workflow/workflow_properties_filter.ts:116]: nit: this is duplicated, maybe move to a shared function?

File: packages/jira-adapter/src/filters/workflow/workflow_properties_filter.ts [hunk 3]
```diff
@@ -46,68 +48,90 @@ const convertStatusesPropertiesToMap = (instance: WorkflowInstance): void => {
...
+
+      const changesToReturn = await awu(relevantChanges)
+        .map(async change => {
+          originalChanges[getChangeData(change).elemID.getFullName()] = change
+          return applyFunctionToChangeData<Change<WorkflowInstance>>(
+            change,
+            async instance => {
+              // I have to call resolveValues here because after I change the status
+              // properties structure the getLookUpName won't know how to resolve
+              // the references inside correctly
```
Comment (netama) [inline packages/jira-adapter/src/filters/workflow/workflow_properties_filter.ts:99]: let's discuss this one IRL? not sure I fully follow

File: packages/jira-adapter/src/reference_mapping.ts [hunk 4]
```diff
@@ -468,6 +472,16 @@ ReferenceContextStrategyName
     serializationStrategy: 'id',
     target: { typeContext: 'parentFieldType' },
   },
+  {
+    src: { field: 'value', parentTypes: ['StatusProperty'] },
+    serializationStrategy: 'id',
+    target: { typeContext: 'workflowStatusPropertiesIdContext' },
+  },
+  {
+    src: { field: 'value', parentTypes: ['StatusProperty'] },
```
Comment (netama) [inline packages/jira-adapter/src/reference_mapping.ts:481]: maybe we can somehow consolidate the two rules if we do have a rule for which serialization should be used? (might be easier to discuss IRL)

File: packages/jira-adapter/src/filters/workflow/workflow_properties_filter.ts (hunk 3 above)
Comment (alonstern) [reply packages/jira-adapter/src/filters/workflow/workflow_properties_filter.ts:99]: Opened https://salto-io.atlassian.net/browse/SALTO-2704

File: packages/jira-adapter/src/reference_mapping.ts (hunk 4 above)
Comment (netama) [reply packages/jira-adapter/src/reference_mapping.ts:481]: discussed IRL, I forgot this is good now 🤦‍♀️

**Review by netama — APPROVED** [review]

**Pull request merged**